src/fillTank.test.js

```diff
@@ -1,11 +1,109 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should be declared', () => {
+    expect(fillTank)
+      .toBeInstanceOf(Function);
+  });
+
+  it(`should order full tank if 'amount' is not given`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const expectedResult = 40;
+
+    fillTank(customer, 4);
+
+    expect(customer.vehicle.fuelRemains).toBe(expectedResult);
+  });
+
+  it(`should order full tank if 'amount'
+    is greater than the tank can accommodate`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const expectedResult = 40;
+
+    fillTank(customer, 4, 90);
+
+    expect(customer.vehicle.fuelRemains).toBe(expectedResult);
+  });
+
+  it(`should fill in only what the client can pay`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const expectedResult = {
+      money: 0,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 13,
+      },
+    };
+
+    fillTank(customer, 600, 30);
 
+    expect(customer).toEqual(expectedResult);
   });
 
-  // write tests here
+  it(`should round the 'fuelRemains' by discarding
+    number to the tenth part`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const expectedResult = 13.5;
+
+    fillTank(customer, 8, 5.599);
```

Comment:
The test case aims to check rounding behavior, but the 'amount' parameter is 8, which might not lead to a fractional fuel remains value. Ensure that the parameters and expected result align with the rounding logic you intend to test.